test/test_HPE3ParClient_volume.py

```diff
@@ -2288,6 +2289,111 @@ def test36_resume_schedule(self):
+        self.cl.createVolume(VOLUME_NAME2, CPG_NAME2, 16384, optional)
+
+        # check
+        vol2 = self.cl.getVolume(VOLUME_NAME2)
+        self.assertIsNotNone(vol2)
+
+        volName = vol2['name']
+        comment = vol2['comment']
+        reduced = vol2['reduce']
+
+        self.assertEqual(VOLUME_NAME2, volName)
+        self.assertEqual("test volume2", comment)
+        self.assertEqual(True, reduced)
+        self.printFooter('create_volume')
+
+    def test_39_offline_copy_volume_merlin_support(self):
+        self.printHeader('copy_volume')
+        self.cl.merlin_supported = True
+        # add one
+        optional = {'comment': 'test volume', 'tpvv': True,
+                    'snapCPG': CPG_NAME1}
+        self.cl.createVolume(VOLUME_NAME1, CPG_NAME1, 1024, optional)
+        self.cl.createVolume(VOLUME_NAME2, CPG_NAME1, 1024, optional)
+        # copy it
+        optional1 = {'online': False}
+        self.cl.copyVolume(VOLUME_NAME1, VOLUME_NAME2, CPG_NAME1, optional1)
+        vol2 = self.cl.getVolume(VOLUME_NAME2)
+        volName = vol2['name']
+        self.assertEqual(VOLUME_NAME2, volName)
+        self.printFooter('copy_volume')
+
+    def test_40_online_copy_volume_merlin_support(self):
+        self.printHeader('copy_volume')
+        self.cl.merlin_supported = True
+        # TODO: Add support for ssh/stopPhysical copy in mock mode
+        if self.unitTest:
+            self.printFooter('copy_volume')
+            return
+        # add one
+        optional = {'comment': 'test volume', 'tpvv': True,
+                    'snapCPG': CPG_NAME1}
+        self.cl.createVolume(VOLUME_NAME1, CPG_NAME1, SIZE, optional)
+
+        # copy it
+        # for online copy we need to specify the tpvv/reduce for merlin
+        optional = {'online': True, 'tpvv': True}
+        self.cl.copyVolume(VOLUME_NAME1, VOLUME_NAME2, CPG_NAME1, optional)
+        vol2 = self.cl.getVolume(VOLUME_NAME2)
+        volName = vol2['name']
+        self.assertEqual(VOLUME_NAME2, volName)
+
+    def test_41_copy_volume_interrupted_merlin_support(self):
+        self.printHeader('copy_volume')
+        self.cl.merlin_supported = True
+        # TODO: Add support for ssh/stopPhysical copy in mock mode
+        if self.unitTest:
+            self.printFooter('copy_volume')
+            return
+        # add one
+        optional = {'comment': 'test volume', 'tpvv': True,
+                    'snapCPG': CPG_NAME1}
+        self.cl.createVolume(VOLUME_NAME1, CPG_NAME1, SIZE, optional)
+
+        # copy it
+        optional = {'online': True, 'tpvv': True}
+        self.cl.copyVolume(VOLUME_NAME1, VOLUME_NAME2, CPG_NAME1, optional)
+        self.cl.getVolume(VOLUME_NAME2)
+        self.cl.stopOnlinePhysicalCopy(VOLUME_NAME2)
+
+        self.assertRaises(
+            exceptions.HTTPNotFound,
+            self.cl.getVolume,
+            VOLUME_NAME2
+        )
+
+        self.printFooter('copy_volume')
+
```

Comment:
If you agree with the truth table attached above, it would be nice to have unit tests based on the input values mentioned in the truth table.

Reply:
Added test cases as per the input values mentioned in the truth table